src/rev_ai/apiclient.py

```diff
@@ -195,19 +193,30 @@ def get_transcript_text(self, id_, filename=None, filepath=None):
         response = self.session.get(url, headers={'Accept': 'text/plain'})
         response.raise_for_status()
 
-        if filename:
-            path = os.path.join(filepath or '', filename+'.txt')
-            with open(path, 'w+') as f:
-                f.write(response.text)
-
         return response.text
 
-    def get_transcript_json(self, id_, filename=None, filepath=None):
-        """Get the transcript of a specific job as json
+    def get_transcript_text_as_stream(self, id_):
+        """Get the transcript of a specific job as a plain text stream.
+
+        :param id_: id of job to be requested
+        :returns: requests.models.Response HTTP response which can be used to stream
+            the payload of the response
+        :raises: HTTPError
+        """
+        if not id_:
+            raise ValueError('id_ must be provided')
```

Comment:
do we need to note this `:raises: ValueError`? What is the convention?

Reply:
It's not strictly necessary. My opinion is that it's a good convention to have just to make it clear what arguments are needed and have better error handling if a customer messes up and passes us something undefined